src/main/java/org/terracotta/statistics/AbstractOperationStatistic.java

```diff
@@ -63,9 +65,22 @@ public long sum() {
   public void begin() {
     if (!derivedStatistics.isEmpty()) {
       long time = Time.time();
+      operationStartTime.set(time);
       for (ChainedOperationObserver<? super T> observer : derivedStatistics) {
         observer.begin(time);
       }
     }
   }
+
+  @Override
+  public void end(T result) {
+    if (!derivedStatistics.isEmpty()) {
+      long time = Time.time();
+      long latency = Math.max(0, time - operationStartTime.get());
```

Comment:
I means latencies are not scoped by outcome?

Reply:
yes they are: T result _is_ the outcome ;-)